backends/postgres.go

```diff
@@ -132,12 +124,32 @@ func NewPostgres(authOpts map[string]string, logLevel log.Level, hasher hashing.
 	//Build the dsn string and try to connect to the db.
 	connStr := fmt.Sprintf("user=%s password=%s dbname=%s host=%s port=%s", postgres.User, postgres.Password, postgres.DBName, postgres.Host, postgres.Port)
 
-	if (postgres.SSLMode == "verify-ca" || postgres.SSLMode == "verify-full") && checkSSL {
-		connStr = fmt.Sprintf("%s sslmode=verify-ca sslcert=%s sslkey=%s sslrootcert=%s", connStr, postgres.SSLCert, postgres.SSLKey, postgres.SSLRootCert)
-	} else if postgres.SSLMode == "require" {
-		connStr = fmt.Sprintf("%s sslmode=require", connStr)
-	} else {
+	switch postgres.SSLMode {
+	case "disable":
 		connStr = fmt.Sprintf("%s sslmode=disable", connStr)
+
+	case "require":
+		connStr = fmt.Sprintf("%s sslmode=require", connStr)
+
+	case "verify-ca":
+		connStr = fmt.Sprintf("%s sslmode=verify-ca", connStr)
+
+	case "verify-full":
+		fallthrough
+	default:
+		connStr = fmt.Sprintf("%s sslmode=verify-full", connStr)
```

Comment:
I do understand that's not a great default and it'd be better to fall to the side of safety so maybe we should indeed change it, but that breaks expectations and would need a bump of the major version. We're at 1.9.1, so maybe it is time we went for 2.0.
Thoughts?

Reply:
The default value of `postgres.SSLMode` is still `disable`, even when there is no `auth_opt_pg_sslmode` set in the config:

https://github.com/iegomez/mosquitto-go-auth/blob/dea95934104aa385e2b02e2b84c3bc7febc364b4/backends/postgres.go#L47-L55

From there it is no breaking change for the moment, no matter what configuration you have.

I think the next step when going to 2.0 could be the change of the default to `verify-full`:

```golang
	var postgres = Postgres{
		Host:           "localhost",
		Port:           "5432",
		SSLMode:        "verify-full",
		SuperuserQuery: "",
		AclQuery:       "",
		hasher:         hasher,
		connectTries:   -1,
	}
```

Reply:
The problem is that we set whatever value is given by the user here without checking if it's allowed: https://github.com/iegomez/mosquitto-go-auth/blob/dea95934104aa385e2b02e2b84c3bc7febc364b4/backends/postgres.go#L101

So you're right passing no option defaults to `disable`, but passing a wrong one now defaults to `verify-full` instead.

Reply:
I understand your concerns. When a user has a complete wrong sslmode the PR breaks his insecure database connection. On the other hand no one is able to use a secure database connection at the moment, which is also quit hard for a authentication plugin. :smile: 

But to move forward I can change the order of the `disable` case so that it is default again.

Reply:
Let's merge with the insecure default for now and then work on defaulting to safety in a different PR.

Reply:
Is there already a branch for the next major (2.0) where I can place such a PR?